pkg/util/https/server.go

```diff
@@ -16,50 +16,111 @@ package https
 
 import (
 	"context"
+	cryptotls "crypto/tls"
 	"net/http"
+	"sync"
+	"time"
 
+	"agones.dev/agones/pkg/util/fswatch"
 	"agones.dev/agones/pkg/util/runtime"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	tlsDir = "/certs/"
+)
+
 // tls is a http server interface to enable easier testing
 type tls interface {
 	Close() error
 	ListenAndServeTLS(certFile, keyFile string) error
 }
 
+// CertServer holds the Server certificate
+type CertServer struct {
+	Certs  *cryptotls.Certificate
+	CertMu sync.Mutex
```

Comment:
nit: these fields doesn't have to exported (beginning with upper case) since they are only used in this package, right? Better to change them to unexported for stricter restriction. Similar for the `CertServer` in `Server`.

Reply:
Yeah! Thanks for pointing. It should be unexported type.

Reply:
Yeah! Thanks for pointing. It should be unexported type.